WireDomain/Sources/WireDomain/Repositories/FeatureConfigRepository.swift

```diff
@@ -0,0 +1,352 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+import WireAPI
+import WireDataModel
+
+/// Facilitates access to feature configs related domain objects.
+
+protocol FeatureConfigRepositoryProtocol {
+
+    /// Pulls feature configs from the server and stores them locally.
+    ///
+    /// The `AsyncStream` produces a `FeatureState` value right after the feature is stored locally.
+    /// so actions can be triggered right away at a lower level callsite (e.g an interactor).
+    /// This will allow a user to be notified immediately even if the stream fails afterwards.
+    ///
+    /// - Returns: An async stream of `FeatureState`.
+
+    func pullFeatureConfigs() -> AsyncThrowingStream<FeatureState, Error>
+
+    /// Fetches a feature config locally.
+    ///
+    /// - Parameter name: The feature name to fetch the config for.
+    /// - Parameter type: The type of config to retrieve.
+    /// - Returns: A `LocalFeature` object with a status and a config (if any).
+
+    func fetchFeatureConfig<T: Decodable>(withName name: Feature.Name, type: T.Type) async throws -> LocalFeature<T>
+
+    /// Fetches a flag indicating whether the user should be notified of a given feature.
+    /// - Parameter name: The feature name.
+    /// - Returns: `true` if user should be notified.
+
+    func fetchNeedsToNotifyUser(forFeatureName name: Feature.Name) async throws -> Bool
+
+    /// Stores a flag indicating whether the user should be notified of a given feature.
+    /// - Parameter notifyUser: Whether the user should be notified for a given feature.
+    /// - Parameter name: The name of the feature to set the flag for.
+
+    func storeNeedsToNotifyUser(_ notifyUser: Bool, forFeatureName name: Feature.Name) async throws
```

Comment:
Yes I guess it could be a good way to regroup all kind of changes into a single category although I think we would still need to handle these alerts from our repositories. This is a just a thought but we could have for instance a BaseRepository (base class for all repositories) that would continuously check for alerts and propagate them to their corresponding interactors and then display them to the user.